src/Signing/SignatureVerifier.php

```diff
@@ -47,12 +55,20 @@ public function __construct(KeyLoader $keyLoader)
      */
     public function hasValidSignature(AuthnRequest $request, ServiceProvider $serviceProvider)
     {
+        $this->logger->debug(sprintf('Extracting public keys for ServiceProvider "%s"', $serviceProvider->getEntityId()));
+
         $keys = $this->keyLoader->extractPublicKeys($serviceProvider);
 
+        $this->logger->debug(sprintf('Found "%d" keys, filtering the keys to get X509 keys', $keys->count()));
         $x509Keys = $keys->filter(function (SAML2_Certificate_Key $key) {
             return $key instanceof SAML2_Certificate_X509;
         });
 
+        $this->logger->debug(sprintf(
+            'Found "%d" X509 keys, attempting to use each for signature verification',
+            $x509Keys->count()
+        ));
+
         foreach ($x509Keys as $key) {
             if ($this->isSignedWith($request, $key)) {
                 return true;
```

Comment:
I'd say a debug message informing the signature was (not) properly signed with one of the keys is useful and completes the log messages. The consumer can still log its process on INFO or whatever (we're throwing if signature validation fails) at its discretion.